File: contactos.py
def solicitar_datos():
    contactos = []
    while True:
        nombre = input("Ingrese nombre: ")
        apellido = input("Ingrese apellido: ")
        dni = input("Ingrese DNI: ")
        
        telefonos = []
        while True:
            telefono = input("Ingrese un teléfono (o presione Enter para finalizar): ")
            if telefono:
                telefonos.append(telefono)
            else:
                break
        
        contactos.append([nombre, apellido, dni, telefonos])
        
        continuar = input("¿Desea ingresar otra persona? (si/no): ")
        if continuar.lower() != 'si':
            break
    
    print("\nLista de contactos:")
    for contacto in contactos:
        print(contacto)

solicitar_datos()

File: contactos.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let contactos = [];

function solicitarDatos() {
    rl.question("Ingrese nombre: ", function(nombre) {
        rl.question("Ingrese apellido: ", function(apellido) {
            rl.question("Ingrese DNI: ", function(dni) {
                let telefonos = [];
                
                function agregarTelefono() {
                    rl.question("Ingrese un teléfono (o presione Enter para finalizar): ", function(telefono) {
                        if (telefono) {
                            telefonos.push(telefono);
                        } else {
                            contactos.push([nombre, apellido, dni, telefonos]);
                            rl.question("¿Desea ingresar otra persona? (si/no): ", function(respuesta) {
                                if (respuesta.toLowerCase() === 'si') {
                                    solicitarDatos();
                                } else {
                                    console.log("\nLista de contactos:");
                                    console.log(contactos);
                                    rl.close();
                                }
                            });
                        }
                    });
                }
                agregarTelefono();
            });
        });
    });
}

solicitarDatos();